pkg/operator/resourcesynccontroller/interfaces.go

```diff
@@ -25,10 +32,22 @@ var (
 
 // ResourceSyncer allows changes to syncing rules by this controller
 type ResourceSyncer interface {
-	// SyncConfigMap indicates that a configmap should be copied from the source to the destination.  It will also
-	// mirror a deletion from the source.  If the source is a zero object the destination will be deleted.
+	// SyncConfigMap indicates that a configmap should be copied from the source
+	// to the destination.  It will also mirror a deletion from the source.  If
+	// the source is a zero object the destination will be deleted.
 	SyncConfigMap(destination, source ResourceLocation) error
-	// SyncSecret indicates that a secret should be copied from the source to the destination.  It will also
-	// mirror a deletion from the source.  If the source is a zero object the destination will be deleted.
+	// SyncConfigMapConditionally conditionally indicates that a configmap should
+	// be copied from the source to the destination.  It will also mirror a
+	// deletion from the source.  If the source is a zero object the destination
+	// will be deleted.
+	SyncConfigMapConditionally(destination, source ResourceLocation, precopreconditionsFulfilledFn PreconditionsFulfilled) error
+	// SyncSecret indicates that a secret should be copied from the source to the
+	// destination.  It will also mirror a deletion from the source.  If the
+	// source is a zero object the destination will be deleted.
 	SyncSecret(destination, source ResourceLocation) error
+	// SyncSecretCondtionally conditionally indicates that a secret should be
+	// copied from the source to the destination.  It will also mirror a deletion
+	// from the source.  If the source is a zero object the destination will be
+	// deleted.
+	SyncSecretConditionally(destination, source ResourceLocation, precopreconditionsFulfilledFn PreconditionsFulfilled) error
```

Comment:
Would it make sense to pass a precondition function during the `NewResourceSyncController` construction? 

We wouldn't have to change the interface, and the precondition would apply to all resources.

Reply:
If we do that, we won't be able to add preconditions per resource, so we would have to create a new resource sync controller inside of the encryption controller instead of reusing the common one.